.github/workflows/release.yml

```diff
@@ -0,0 +1,556 @@
+        --detach-sig build/linux/meson-dist/taisei-v${{ steps.vars.outputs.tag }}.tar.xz
+
+    - name: Upload Artifact (Source)
+      uses: actions/upload-artifact@v2
+      if: success()
+      with:
+        name: source-${{ steps.vars.outputs.tag }}-txz
+        path: build/linux/meson-dist/taisei-v${{ steps.vars.outputs.tag }}.tar.xz*
+        if-no-files-found: error
+
+    - name: Upload Log
+      uses: actions/upload-artifact@v2
+      with:
+        name: taisei_x64_linux_release_log
+        path: build/linux/meson-logs/meson-log.txt
+        if-no-files-found: warn
+
+  macos-release-build-universal:
+    name: macOS (Universal)
+    runs-on: macos-11
+    steps:
+    - name: Install Python
+      uses: actions/setup-python@v2
+      with:
+        python-version: '3.x'
+
+    - name: Install Tools
+      run: >
+        brew install
+        gcc
+        pkg-config
+        docutils
+        pygments
+        create-dmg
+        gnupg
+
+        pip install
+        meson==${{ env.MESON_VERSION }}
+        ninja
+        zstandard
+        python-gnupg
+
+    - name: Checkout Code
+      uses: actions/checkout@v2
+      with:
+        submodules: "recursive"
+        fetch-depth: 1
+
+    - name: Import GPG Key
+      uses: taisei-project/ghaction-import-gpg@v3
+      with:
+        gpg-private-key: ${{ secrets.GPG_PRIVATE_KEY }}
+        passphrase: ${{ secrets.GPG_PASSPHRASE }}
+
+    - name: Set Build Env
+      id: vars
+      run: |
+        git fetch --force --tags
+        echo ::set-output name=tag::${GITHUB_REF#refs/tags/v}
+
+    - name: Configure & Build
+      run: >
+        $(pwd)/scripts/macos_build_universal.sh $(pwd)/build-release/ $(pwd)
+      shell: bash
+
+    - name: Run Test
+      run: $(pwd)/build-release/compiled/combined/Taisei.app/Contents/MacOS/Taisei -R $(pwd)/misc/ci/tests/test-replay.tsr
+      env:
+        TAISEI_NOPRELOAD: ${{ env.TAISEI_NOPRELOAD }}
+        TAISEI_PRELOAD_REQUIRED: ${{ env.TAISEI_PRELOAD_REQUIRED }}
+
+    - name: Upload Artifact
+      uses: actions/upload-artifact@v2
+      if: success()
+      with:
+        name: macos-universal-${{ steps.vars.outputs.tag }}-dmg
+        path: build-release/compiled/Taisei-${{ steps.vars.outputs.tag }}-universal.dmg*
+        if-no-files-found: error
+
+    - name: Upload Log
+      uses: actions/upload-artifact@v2
+      with:
+        name: taisei_macos_universal_release_log
+        path: build-release/*/meson-logs/meson-log.txt
+        if-no-files-found: warn
+
+  windows-release-build-x64:
+    name: Windows (x64)
+    runs-on: ubuntu-20.04
+    container: mstorsjo/llvm-mingw:20210423 # cross-compiler with mingw
+    steps:
+    - name: Install Tools
+      run: >
+        apt update || true
+
+        apt install -y -qq software-properties-common
+
+        add-apt-repository ppa:git-core/ppa -y
+
+        apt install -y -qq
+        python3-docutils
+        python3-pip
+        git
+        wget
+        nsis
```

Comment:
I think I originally did this so that every dependency that `nsis` depended on was installed before doing the raw `dpkg -i`, but I'll double-check that.